Batteries/Util/LibraryNote.lean

```diff
@@ -35,6 +35,45 @@ creates a new "library note", which can then be cross-referenced using
 -- See note [some tag]
 ```
 in doc-comments.
+Use `#help note "some tag"` to display all notes with the tag
+`"some tag"` in the infoview.
 -/
 elab "library_note " title:strLit ppSpace text:docComment : command => do
   modifyEnv (libraryNoteExt.addEntry · (title.getString, text.getDocString))
+
+
+open Lean Parser in
+/--
+`#help note "foo"` searches (case-insensitively) for all library notes whose
+label starts with "foo", then displays those library notes sorted alphabetically by label,
+grouped (case-sensitively) by label.
+The command only displays the library notes that are declared in
+imported files or in the same file above the line containing the command.
+-/
+elab "#help note" name:strLit : command => do
+  let env ← getEnv
+
+  -- get the library notes from both this and imported files
+  let local_entries := (libraryNoteExt.getEntries env).reverse
+  let imported_entries := (libraryNoteExt.toEnvExtension.getState env).importedEntries
+
+  -- filter for the appropriate notes while casting to list
+  let label_prefix := name.getString.toLower
+  let imported_entries_filtered := imported_entries.flatten.toList.filterMap
+    (fun x => if label_prefix.isPrefixOf x.fst.toLower then some x else none)
+  let valid_entries := imported_entries_filtered ++ local_entries.filterMap
+    (fun x => if label_prefix.isPrefixOf x.fst.toLower then some x else none)
+
+  let grouped_valid_entries := valid_entries.mergeSort (·.fst ≤ ·.fst)
+    |>.groupBy (·.fst == ·.fst)
+
+  -- display results in a readable style
+  if grouped_valid_entries.isEmpty then
+    logInfo "Note not found"
+  else
+    logInfo <| "\n\n".intercalate <|
+      grouped_valid_entries.map (fun l =>
+        "library_note \"" ++ (@List.head! _ ⟨⟨"",""⟩⟩ l
+          ).fst ++ "\"\n"
+        ++ ("\n\n".intercalate <| l.map (fun e => "/--\n" ++ e.snd.trim ++ "\n-/")))
```

Comment:
Add `deriving Inhabited` after the definition of `LibraryNoteEntry` to make this work:
```suggestion
    logInfo <| "\n\n".intercalate <|
      grouped_valid_entries.map fun l =>
        "library_note \"" ++ l.head!.fst ++ "\"\n" ++ 
        ("\n\n".intercalate <| l.map fun e => "/--\n" ++ e.snd.trim ++ "\n-/")
```